news-bundle/src/Migration/FeedMigration.php

```diff
@@ -0,0 +1,107 @@
+<?php
+
+declare(strict_types=1);
+
+/*
+ * This file is part of Contao.
+ *
+ * (c) Leo Feyer
+ *
+ * @license LGPL-3.0-or-later
+ */
+
+namespace Contao\NewsBundle\Migration;
+
+use Contao\CoreBundle\Migration\AbstractMigration;
+use Contao\CoreBundle\Migration\MigrationResult;
+use Doctrine\DBAL\Connection;
+use Psr\Log\LoggerInterface;
+
+class FeedMigration extends AbstractMigration
+{
+    public function __construct(private readonly Connection $connection, private readonly LoggerInterface $logger)
+    {
+    }
+
+    public function shouldRun(): bool
+    {
+        if (!$this->connection->createSchemaManager()->tablesExist(['tl_news_feed'])) {
+            return false;
+        }
+
+        $result = $this->connection->executeQuery('SELECT COUNT(id) AS count FROM tl_news_feed')->fetchFirstColumn();
+
+        return !empty($result) && $result[0] > 0;
+    }
+
+    public function run(): MigrationResult
+    {
+        // Add new columns to `tl_page`
+        $newFields = [
+            'newsArchives' => 'blob NULL',
+            'feedFormat' => "varchar(32) NOT NULL default 'rss'",
+            'feedSource' => "varchar(32) NOT NULL default 'source_teaser'",
+            'maxFeedItems' => 'smallint(5) unsigned NOT NULL default 25',
+            'feedFeatured' => "varchar(16) COLLATE ascii_bin NOT NULL default 'all_items'",
+            'imgSize' => "varchar(255) NOT NULL default ''",
+        ];
+
+        foreach ($newFields as $field => $definition) {
+            $this->connection->executeStatement("
+                ALTER TABLE
+                    tl_page
+                ADD
+                    $field $definition
+            ");
```

Comment:
Either the `shouldRun` needs to make sure the migration is only executed if these fields don't exist, or better we should probably check for the fields here and skip existing ones.

Reply:
Fixed in ccc498110